### DISCUSSION TITLE
Flask-Session sessions not storing when the session is changed inside a websocket

### DISCUSSION THREAD

**Discussion by chronosirius** [miguelgrinberg/Flask-SocketIO#1859] [Q&A]
I have the user press a button and it is supposed to update the session, but it doesn't seem to work. I'm not sure if this is a Flask-SocketIO problem or a Flask-Session, so I am asking here and there. I updated the session inside the websocket event but later when I try to get information from the session from an HTTP request, it isn't there.

**Comment by miguelgrinberg** [comment]
If you are using Flask 2.2, try a downgrade to 2.1. There is an open issue with Flask 2.2 that I'm currently looking at. See https://github.com/miguelgrinberg/Flask-SocketIO/issues/1856

**Reply by chronosirius** [reply]
I am using Flask 2.1.2

**Reply by miguelgrinberg** [reply]
Then try the sessions.py example in this repository, which demonstrates how to do it. I have been using this example to troubleshoot the 2.2 problem, so I'm positive that example works well for the latest Flask-Session and Flask < 2.2.

**Reply by chronosirius** [reply]
I compared the sessions.py example to my current code, but I am doing it the same way. My Flask-Session is up-to-date with the PyPI release.

**Reply by miguelgrinberg** [reply]
Then please show me how sessions are broken with my code, not with yours. If they apps are similar then my code must be also broken, correct?

**Reply by chronosirius** [reply]
In the example, is the client sending the set-session after receiving the refresh-session event? Because if so then that is not how I am doing it, I am changing it on the fly. If it helps, here is the snippet of my code that is not working.
```py
session['li'] = True
session['id'] = account['id']
session['username'] = account['username']
session['email'] = data['email']
session['discriminator'] = users[account['id']]['discriminator']
session.modified = True
print(session)
```
The output results in just a Python dictionary, but in the http requests, it shows a <FileSystemSession (dict)>.

**Reply by miguelgrinberg** [reply]
Did you set `manage_session=False`? If you are getting a dictionary you are using the internal session management, not Flask's.

**Reply by chronosirius** [reply]
No, I did not. I have upgraded Flask-SocketIO to 5.3.0 and Flask to 2.2.2, and it still is not working.

**Reply by miguelgrinberg** [reply]
The ` manage_session` setting should be False in your case, as I indicated above. See the documentation if you need more information.

**Reply by chronosirius** [reply]
Flask(..., manage_session=False) raises "Flask.__init__() got an unexpected argument: 'manage_session'"

**Reply by miguelgrinberg** [reply]
I suggest you work with the session example first, until you become familiar with how this works: https://github.com/miguelgrinberg/Flask-SocketIO/blob/main/example/sessions.py.

**Reply by chronosirius** [reply]
Sorry, it seems I had misunderstood you earlier. The SocketIO(app) should have manage_session, not Flask. I found [this](https://blog.miguelgrinberg.com/post/flask-socketio-and-the-user-session) which was very helpful. I will try this at my next opportunity.

**Reply by chronosirius** [reply]
It didn't work.

**Reply by miguelgrinberg** [reply]
My sessions example doesn't work? How so?

**Reply by chronosirius** [reply]
Your sessions example worked on replit (I didn't try my env), but setting `manage_session=False` did not.

**Reply by miguelgrinberg** [reply]
I can't really help you if you don't provide any information except that "it doesn't work". Do we agree that this is likely a problem in your code and not a bug? Why don't you post a complete example that shows the problem?

**Reply by chronosirius** [reply]
Well, I'm trying to build a login page with websockets, (without Flask-Login) and I store the details in the session, but as soon as they are redirected to the page I want them to go to, it shows that the session didn't persist and redirects me back to the login page, like any good website should. I have tested the same code on replit and locally, and in both cases the code failed to store my session. However, I tried your session example on replit and it worked fine. That identified that there is a problem in my code and not my environment.

My code:
https://github.com/chronosirius/DisDatCorddd/blob/master/main.py
https://github.com/chronosirius/DisDatCorddd/blob/master/templates/usr/login.html

**Reply by miguelgrinberg** [reply]
What do you mean by "redirected"? Your Socket.IO connection will break if you redirect to another page.

**Reply by chronosirius** [reply]
the client-side recieves the login-success event which is sent after the session is updated, and redirects the user to the logged in page.

**Reply by miguelgrinberg** [reply]
I don't know how to run your application. From looking at it I cannot tell what's wrong, except that using Socket.IO when you use redirects does not really make sense, and my guess is that the issue is somehow related. My recommendation is that you debug your application specifically looking at when the browser navigates to a new page and breaks the Socket.IO connection.

**Reply by chronosirius** [reply]
How would I do that?

**Reply by chronosirius** [reply]
I found a solution by adding a random session key-value pair in the view function for the login page.
